scheduler/cloudformation/cloudformation.go

```diff
@@ -280,13 +294,118 @@ func (s *Scheduler) submit(ctx context.Context, tx *sql.Tx, app *scheduler.App,
 	return nil
 }
 
-// updateStack performs a stack update, but waits for the stack to enter a
-// stable state before starting.
+func (s *Scheduler) createStack(input *cloudformation.CreateStackInput, done chan error) error {
+	p := func() error {
+		_, err := s.cloudformation.CreateStack(input)
+		return err
+	}
+	waiter := s.cloudformation.WaitUntilStackCreateComplete
+	locked := make(chan struct{})
+	return s.performStackOperation(*input.StackName, p, waiter, locked, done)
+}
+
+// updateStack enqueues the stack update, and waits a set amount of time for the
+// stack update to be submitted.
+func (s *Scheduler) updateStack(input *cloudformation.UpdateStackInput, done chan error) error {
+	locked := make(chan struct{})
+	errCh := make(chan error)
+
+	go func() {
+		p := func() error { return s.executeStackUpdate(input) }
+		waiter := s.cloudformation.WaitUntilStackUpdateComplete
+		errCh <- s.performStackOperation(*input.StackName, p, waiter, locked, done)
+	}()
+
+	var err error
+	select {
+	case <-time.After(lockWait):
+		// FIXME: At this point, we don't want to affect UX by waiting
```

Comment:
Yeah, definitely. I was thinking we could change the interface of `scheduler.Submit` to take a `done chan error` arg. Empire could then use this to submit quickly, but receive on that channel to be notified when it was actually fullfilled (e.g. the stack update completed).

Then we could expose this in the `emp releases` command (as well as events), so you can tell if a release is pending, completed, canceled, etc.

But yeah, probably a separate PR.